test/core/providers/ethereum.ts

```diff
@@ -0,0 +1,85 @@
+import { assert } from "chai";
+import { EventEmitter } from "events";
+import { EthereumProvider } from "../../../src/core/providers/ethereum";
+import {
+  Callback,
+  JsonRPCRequest,
+  JsonRPCResponse,
+  Provider
+} from "web3x/providers";
+import { toPayload } from "web3x/request-manager/jsonrpc";
+
+class MockedHttpProvider extends EventEmitter implements Provider {
+  constructor(host: string) {
+    super();
+  }
+  public send(payload: JsonRPCRequest, callback: Callback) {
+    if (payload.method === "net_version") {
+      callback(undefined, {
+        jsonrpc: "2.0",
+        id: 1,
+        result: "4"
+      });
+    } else if (payload.method === "bleep") {
+      callback(new Error("Method not found"), undefined);
+    } else if (payload.method === "fail_method") {
+      callback(undefined, {
+        jsonrpc: "2.0",
+        id: 2,
+        result: {
+          error: "do not meet the requirements",
+          code: -32345
+        }
+      });
+    } else {
+      callback(undefined, payload);
```

Comment:
This makes the test really confusing, as you are returning a response as request. You could return `payload. params`.